### PR TITLE
Fixes #18669: Properly locate entity errors if the source has metadatas attached

### PR DESCRIPTION
**Pull request by gpoblon**
https://issues.rudder.io/issues/18669

### CODE REVIEW

**Comment by gpoblon** [comment]
test command `cargo run -- compile -c tools/rudderc-dev.conf -i tests/techniques/simplest/technique.rl -f cf`

File: rudder-lang/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (amousset) [inline rudder-lang/.vscode/launch.json:1]: .vscode folder should be in gitignore

**Comment by peckpeck** [comment]
you should do a fixup and not an amend as soon as someone has started reviewing the code

**Comment by peckpeck** [comment]
the merge bot does an autosquash wich merges all fixups into a single commit

File: rudder-lang/src/parser/tests.rs [hunk 2]
```diff
@@ -259,14 +264,36 @@ fn test_penum() {
             }
         ))
     );
+    assert_eq!(
+        // to test preceeding megtadata, directly test a pdeclaration
+        map_res(penum, "enum abc3 { @metadata=\"value\"\n a, b, }"),
```
Comment (peckpeck) [inline rudder-lang/src/parser/tests.rs:269]: metaada should only be parsed at the begining of a line

File: rudder-lang/src/ir/ir1.rs [hunk 3]
```diff
@@ -468,7 +468,7 @@ mod tests {
             "let c2 = case { X => \"val\", Y => \"val2\" }\n",
         );
 
-        parse_str_ok(&mut source, "ext1", "v=if X => \"val\"\n");
+        // parse_str_ok(&mut source, "ext1", "v=if X => \"val\"\n");
```
Comment (peckpeck) [inline rudder-lang/src/ir/ir1.rs:471]: why doesn't this work aymore ?

File: rudder-lang/src/parser/tests.rs (hunk 2 above)
Comment (gpoblon) [reply rudder-lang/src/parser/tests.rs]: fixed :)

File: rudder-lang/src/ir/ir1.rs (hunk 3 above)
Comment (gpoblon) [reply rudder-lang/src/ir/ir1.rs]: forgot to uncomment it, fixed, and test passes

**Review by peckpeck — APPROVED** [review]

**Comment by gpoblon** [comment]
OK, squash merging this PR

**Pull request merged**